lib/components/map/default-map.tsx

```diff
@@ -304,12 +304,12 @@ class DefaultMap extends Component {
           zoom={zoom}
         >
           <PointPopup />
+          <NearbyViewDotOverlay />
```

Comment:
Maplibre has this feature to show the user's current position that could be useful: https://maplibre.org/maplibre-gl-js/docs/examples/locate-user/

Reply:
So yes, I was looking at this, but you can't manually trigger the blue dot. You have to add the control the map, then when the user clicks on it it will show the blue dot. 

Maybe we can do this:
- If a user opens Nearby without coordinates (eg, by clicking on the menu bar icon), we can use the map center as the location.
- While nearby view is open, we show the geolocate control on the map. Then `onGeolocate` we navigate to nearby view with the coordinates specified. 

The downsides:  
- it would be not be very obvious that clicking on the map's geolocate control will change the nearby view to center on that location. 
- when opening the nearby view without coordinates it isn't always clear where the map center is. I guess we could add a map layer that reads the current nearby view coordinates and puts a dot there? How will people know the dot corresponds with nearby view centerpoint?

Reply:
>     * If a user opens Nearby without coordinates (eg, by clicking on the menu bar icon), we can use the map center as the location.
 I think this is necessary.

>     * While nearby view is open, we show the geolocate control on the map. Then `onGeolocate` we navigate to nearby view with the coordinates specified.

Probably for another PR. I think the geolocate control should be shown all the time so that users can find themselves regardless of view.

> The downsides:
> 
>     * it would be not be very obvious that clicking on the map's geolocate control will change the nearby view to center on that location.

Showing the geolocate control all the time would address this downside imo.

Reply:
Then I think we're good here. Part one is done, Nearby View uses the current map coords when it doesn't have anything else to use.